optuna/visualization/_parallel_coordinate.py

```diff
@@ -123,20 +169,30 @@ def _target(t: FrozenTrial) -> float:
     skipped_trial_ids = _get_skipped_trial_numbers(trials, sorted_params)
 
     objectives = tuple([target(t) for t in trials if t.number not in skipped_trial_ids])
+    # The value of (0, 0) is a dummy range. It is ignored when we plot.
+    objective_range = (min(objectives), max(objectives)) if len(objectives) > 0 else (0, 0)
+    dim_objective = _DimensionInfo(label=target_name, values=objectives, range=objective_range)
+
+    if len(trials) == 0:
+        _logger.warning("Your study does not have any completed trials.")
+        return _ParallelCoordinateInfo(
+            dim_objective=dim_objective,
+            dims_params=[],
+            reverse_scale=reverse_scale,
+            target_name=target_name,
+        )
 
     if len(objectives) == 0:
         _logger.warning("Your study has only completed trials with missing parameters.")
-        return go.Figure(data=[], layout=layout)
-
-    dims: List[Dict[str, Any]] = [
-        {
-            "label": target_name,
-            "values": objectives,
-            "range": (min(objectives), max(objectives)),
-        }
-    ]
+        return _ParallelCoordinateInfo(
+            dim_objective=dim_objective,
+            dims_params=[],
+            reverse_scale=reverse_scale,
+            target_name=target_name,
+        )
 
     numeric_cat_params_indices: List[int] = []
+    dims = []
     for dim_index, p_name in enumerate(sorted_params, start=1):
```

Comment:
[nits] you can use start=0 here.

Reply:
It starts 1 because we want to add the dim_objective as 0th dim right after this for block.